qiskit/transpiler/passes/layout/apply_layout.py

```diff
@@ -59,7 +59,7 @@ def run(self, dag):
         for creg in dag.cregs.values():
             new_dag.add_creg(creg)
         for node in dag.topological_op_nodes():
-            qargs = [q[layout[qarg]] for qarg in node.qargs]
+            qargs = [q[layout._v2p[qarg]] for qarg in node.qargs]
```

Comment:
```python
        virtual_physical_map = layout.get_virtual_bits()
        for node in dag.topological_op_nodes():
            qargs = [q[virtual_physical_map[qarg]] for qarg in node.qargs]
```
Can we instead make one call to `Layout` to pre-fetch the `virtual_physical_map`?

Reply:
Yeah, that should work. Done in: https://github.com/Qiskit/qiskit-terra/pull/7036/commits/0154230fa4c260d7942b11f39d1ca07456fa612e